lib/assessment.js

```diff
@@ -38,8 +54,17 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // If the argument obj is unset, create a new object
 // The function you write should return
 // the object passed in as an argument or the object created
-var addMethod = function addMethod(propertyName, method, obj) {
 
+var addMethod = function addMethod(propertyName, method, obj) {
+  if(typeof arguments[2] === 'undefined') {
+      var newObj = {}
+      newObj[propertyName] = method;
+      return newObj;
+  }
+  else  {
+    obj[propertyName] = method;
+    return obj;
+      }
```

Comment:
Nice, Matt!
Much cleaner and better code indeed! And...love the || one too! 
I realized the usage of arguments[2] was definitely not ideal and could lead to error, but couldn't think of a simpler way to do it at the moment. Need to practice more and read more code for things to connect. :)